containers/ecr-viewer/src/app/services/ecrMetadataService.ts

```diff
@@ -96,36 +110,77 @@ export const evaluateEcrMetadata = (fhirBundle: Bundle): EcrMetadata => {
     }
   };
 
-  const fhirERSDWarnings = evaluateAll(
+  const fhirEICRProcessingStatus = evaluateValue(
     fhirBundle,
-    fhirPathMappings.eRSDwarnings,
+    fhirPathMappings.eICRProcessingStatus,
   );
-  const eRSDTextList: ERSDWarning[] = [];
-
-  for (const warning of fhirERSDWarnings) {
-    if (warning.code === "RRVS34") {
-      eRSDTextList.push({
-        warning:
-          "Sending organization is using an malformed eRSD (RCTC) version",
-        versionUsed: "2020-06-23",
-        expectedVersion:
-          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
-        suggestedSolution:
-          "The trigger code version your organization is using could not be determined. The trigger codes may be out date. Please have your EHR administrators update the version format for complete eCR functioning.",
-      });
-    } else if (warning.code === "RRVS29") {
-      eRSDTextList.push({
-        warning:
-          "Sending organization is using an outdated eRSD (RCTC) version",
-        versionUsed: "2020-06-23",
-        expectedVersion:
-          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
-        suggestedSolution:
-          "The trigger code version your organization is using is out-of-date. Please have your EHR administration install the current version for complete eCR functioning.",
-      });
+  const fhirEICRProcessingStatusReasonObs = evaluateOne(
+    fhirBundle,
+    fhirPathMappings.eICRProcessingStatusReason,
+  );
+
+  function geteRSDInfo(
+    processingStatus: string | undefined,
+    reasonObs: Observation | undefined,
+  ): ERSDInfo {
+    if (processingStatus === "RRVS19") {
+      return { success: true };
+    } else if (processingStatus && !reasonObs) {
+      return {
+        success: false,
+        eRSDWarning: {
+          warning: "eICR processed with a warning or error (unknown)",
+          versionUsed: noData,
+          versionExpected: noData,
+          suggestedSolution: noData,
+        },
+      };
+    } else if (!processingStatus && !reasonObs) {
+      return { success: false };
     }
```

Comment:
should this one be undefined? we don't know anything at this point, so have no data